lib/mix/tasks/docs.ex

```diff
@@ -93,9 +93,43 @@ defmodule Mix.Tasks.Docs do
       the "assets" directory in the output path under the name "cover" and the
       appropriate extension. This option has no effect when using the "html" formatter.
 
-    * `:deps` - A keyword list application names and their documentation URL.
-      ExDoc will by default include all dependencies and assume they are hosted on
-      HexDocs. This can be overridden by your own values. Example: `[plug: "https://myserver/plug/"]`
+    * `:deps` - A keyword list with application names and their documentation URL
+      or a function for generating such a list dynamically. ExDoc will by default include
+      all dependencies and assume they are hosted on HexDocs. This can be overridden by
+      your own values.
+
+      Using a keyword list:
+      
+      ```elixir 
+      docs: [
+        deps: [plug: "https://myserver/plug/"]
+      ]
+      ```
+
+      Alternatively you can use a function to generate this list dynamically. For example
+      you could read a custom project option from your dependencies and update the deps
+      list if this is set:
```

Comment:
That's not what I had in mind... but this is better. :)

Actually, you can wrap the whole docs generation in a function, which means we don't need to support functions here. You can do all of the below today by doing this:

```
docs: fn ->
  [deps: deps_base_urls()]
end
```

So I think we can close this? :)

Reply:
Awesome I had missed this, yes we can close this :)